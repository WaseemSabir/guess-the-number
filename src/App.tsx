import "./styles.css";
import React, {useEffect, useState} from 'react';
import GameConfig from "./components/GameConfig";
import TakeGuess from "./components/TakeGuess";
import GameState from "./components/GameState";
import makeStatus from "./utils";

export default function App() {

    // random number between the bounds will be set on start, and each reset click
    const [numberToGuess, setNumberToGuess] = useState(0)
    const [lowerBound, setLowerBound] = useState(1)
    const [upperBound, setUpperBound] = useState(10)
    const [lastGuess, setLastGuess] = useState('none')
    const [status, setStatus] = useState('')

    useEffect(() => {
        // will only run once, on start
        resetNumber()
    }, [])

    function resetNumber() {
        // generate random number within the bound
        let range = upperBound - lowerBound;
        let new_number = Math.trunc((Math.random() * range) + lowerBound);
        setNumberToGuess(new_number)
        setStatus('');
        setLastGuess('none')
    }

    function onNewGuess(guess: number) {
        setLastGuess(guess.toString())
        setStatus(makeStatus(guess, numberToGuess))
    }

    return (
        <>
            <div>
                <GameState
                    lower={lowerBound.toString()}
                    upper={upperBound.toString()}
                    lastGuess={lastGuess}
                    status={status}/>
                <TakeGuess onNewGuess={onNewGuess}/>
            </div>
            <GameConfig
                lowerBound={lowerBound}
                setLowerBound={setLowerBound}
                upperBound={upperBound}
                setUpperBound={setUpperBound}
                resetNumber={resetNumber}/>
        </>
    );
}
