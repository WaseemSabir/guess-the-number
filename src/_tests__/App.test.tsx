import React from "react";
import App from "../App";
import {render, cleanup} from "@testing-library/react";
import renderer, {ReactTestRendererJSON} from 'react-test-renderer';

afterEach(cleanup)

it('should render without crashes', () => {
    const {unmount} = render(<App/>);
    unmount();
});

beforeEach(() => {
    jest.spyOn(global.Math, 'random').mockReturnValue(0.123456789);
});

afterEach(() => {
    jest.spyOn(global.Math, 'random').mockRestore();
});

it("test snapshot with default bound", () => {
    const component = renderer.create(
        <App/>,
    );
    let tree = component.toJSON() as ReactTestRendererJSON;
    expect(tree).toMatchSnapshot();
});
