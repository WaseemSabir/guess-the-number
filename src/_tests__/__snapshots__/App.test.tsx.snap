// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshot with default bound 1`] = `
Array [
  <div>
    <h3>
      Play!
    </h3>
    <div>
      Guess the number between 
      1
       and 
      10
    </div>
    <div>
      Last guess: 
      none
    </div>
    <label>
      Guess: 
    </label>
    <input
      onChange={[Function]}
      onKeyDown={[Function]}
      type="number"
      value=""
    />
    <button
      onClick={[Function]}
    >
      Make guess
    </button>
  </div>,
  <h3>
    Game Config
  </h3>,
  <div>
    <label
      data-testid="lower-bound-label"
    >
      Lower bound: 
    </label>
    <input
      data-testid="lower-bound-input"
      id="lower-bound-input"
      onChange={[Function]}
      type="number"
      value={1}
    />
  </div>,
  <div>
    <label
      data-testid="upper-bound-label"
    >
      Upper bound: 
    </label>
    <input
      data-testid="upper-bound-input"
      id="upper-bound-input"
      onChange={[Function]}
      type="number"
      value={10}
    />
  </div>,
  <button
    data-testid="reset-button"
    onClick={[Function]}
  >
    Reset
  </button>,
]
`;
